Utils/Piece.py

```diff
@@ -1,10 +1,10 @@
 class Piece:
     def __init__(self, position):
         self.position = position
+        self.y, self.x = position
```

Comment:
This is weird. Why duplicating? Use either `x` and `y`, or `position`. Ideally, I would recommend to create a Pydantic model for that like:

```
class Position(BaseModel):
    x: int
    y: int
```